src/app/components/Tokens/hooks.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect, useState } from 'react'
+import { getEthAccountAddressFromBase64, getEvmBech32Address } from '../../utils/helpers'
+import { Layer, useGetRuntimeEvmTokensAddress } from '../../../oasis-nexus/api'
+import { SearchScope } from '../../../types/searchScope'
+import { AppErrors } from '../../../types/errors'
+
+export const useTokenWithBase64Address = (scope: SearchScope, base64Address: string | undefined) => {
+  const [oasisAddress, setOasisAddress] = useState('')
+  const ethAddress = base64Address ? getEthAccountAddressFromBase64(base64Address) : undefined
+  const { network, layer } = scope
+  if (layer === Layer.consensus) {
+    throw AppErrors.UnsupportedLayer
+  }
+  useEffect(() => {
+    if (ethAddress) {
+      getEvmBech32Address(ethAddress).then(setOasisAddress)
+    }
+  }, [ethAddress])
+  const { isLoading, isError, data } = useGetRuntimeEvmTokensAddress(network, layer, oasisAddress, {
+    query: {
+      enabled: !!oasisAddress,
```

Comment:
Yes! According to [the docs](https://tanstack.com/query/latest/docs/react/reference/useQuery), it's true when:

> if there's no cached data and no query attempt was finished yet.

I find it kinda misleading, compared to the natural interpretation of the English naming. That is why I do [this](https://github.com/oasisprotocol/explorer/blob/3a7f7fdddb65f41ceadca0aa4e465dc67a7314e0/src/app/pages/TokenDashboardPage/hook.ts#L25) sometimes.

In the current case, the only case when it would return a false positive is when there is no valid base64 address input, which I have now fixed.